src/bin/dual-iir.rs

```diff
@@ -439,11 +546,30 @@ mod app {
             .settings
             .lock(|settings| (settings.afe, settings.telemetry_period));
 
+        let (pounder_temp, input_powers) =
+            c.shared.pounder_devices.lock(|pounder_dev| {
+                if let Some(dev) = pounder_dev {
+                    let input_powers = [
+                        dev.measure_power(PounderChannel::In0).unwrap(),
+                        dev.measure_power(PounderChannel::In1).unwrap(),
+                    ];
+
+                    (
+                        Some(dev.lm75.read_temperature().unwrap()),
+                        Some(input_powers),
+                    )
+                } else {
+                    (None, None)
+                }
+            });
+
         c.shared.network.lock(|net| {
             net.telemetry.publish(&telemetry.finalize(
                 gains[0],
                 gains[1],
                 c.local.cpu_temp_sensor.get_temperature().unwrap(),
+                pounder_temp,
```

Comment:
No need to shuttle these through `finalize()`. They are not converted with any gains. Continuously appending to the `finalize()` arguments doesn't scale well.
Instead add fields to the `TelemetryBuffer` and let `finalize()` copy everything that doesn't need conversion.
(Same holds for cpu temperature.)

Reply:
[f9459ae](https://github.com/quartiq/stabilizer/pull/610/commits/f9459aee8fdcd0f0ec6efc95cfac0f0d46d9229e)